messybrainz/db/tests/test_recording.py

```diff
@@ -0,0 +1,149 @@
+import json
+from messybrainz import submit_listens_and_sing_me_a_sweet_song as submit_listens
+from messybrainz import db
+from messybrainz.db import data
+from messybrainz.db.testing import DatabaseTestCase
+from messybrainz.db.recording import fetch_distinct_recording_mbids,\
+                                    fetch_gids_for_recording_mbid,\
+                                    link_recording_mbid_to_recording_msid,\
+                                    insert_recording_cluster,\
+                                    create_recording_clusters,\
+                                    get_recording_cluster_id_using_recording_mbid
+
+
+class RecordingTestCase(DatabaseTestCase):
+    def _load_test_data(self, filename):
+        """Loads data for tests from a given JSON file name."""
+
+        with open(self.path_to_data_file(filename)) as f:
+            recordings = json.load(f)
+        
+        msb_listens = []
+        for recording in recordings:
+            messy_dict = {
+                'artist': recording['artist'],
+                'title': recording['title'],
+                'recording_mbid': recording['recording_mbid'],
+            }
+            msb_listens.append(messy_dict)
+        return msb_listens
+
+
+    def test_fetch_distinct_recording_mbids(self):
+        """Tests if recording_mbids are correctly fetched from recording_json table."""
+
+        msb_listens = self._load_test_data('valid_recordings_with_recording_mbids.json')
+        submit_listens(msb_listens)
+        recording_mbids_submitted = {'5465ca86-3881-4349-81b2-6efbd3a59451',
+            '6ba092ae-aaf7-4154-b987-9eb9d05f8616',
+            'cad174ad-d683-4858-a205-7bdc4175fff7',
+        }
+        with db.engine.begin() as connection:
+            recording_mbids_fetched = fetch_distinct_recording_mbids(connection)
+            self.assertEqual(len(recording_mbids_fetched), 3)
+            self.assertSetEqual(recording_mbids_submitted, set(recording_mbids_fetched))
+
+
+    def test_fetch_gids_for_recording_mbid(self):
+        """Tests if gids are correctly fetched."""
+
+        recording_1 = {
```

Comment:
mentioning the difference between the two recordings (& vs and) in a comment would be helpful, took me some time to find. :D